src/abstract/AJSingleVaultTerminal.sol

```diff
@@ -76,15 +76,30 @@ abstract contract AJSingleVaultTerminal is AJPayoutRedemptionTerminal {
         // Check if we need to do a deposit or withdraw to reach the target PPM
         int256 _changeAmount = _targetLocalBalanceDelta(_currentVault, 0);
         if (_changeAmount > 0) {
-            _currentVault.state.localBalance += uint256(_changeAmount);
+            _currentVault.state.localBalance -= uint256(_changeAmount);
+            // max deposit check to avoid tx being reverted
+            uint256 _maxDeposit = _currentVault.impl.maxDeposit(address(this));
+            if (_maxDeposit > 0 && uint256(_changeAmount) > _maxDeposit) {
+                _currentVault.state.shares += _deposit(
```

Comment:
If we are depositing the `_maxDeposit` but accounting for `_changeAmount` (line 79) then the `localBalance` becomes lower than it in reality is, which causes projects to lose funds